features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/Throttling/Subscription/List.jsx

```diff
@@ -0,0 +1,296 @@
+            }),
+            options: {
+                filter: true,
+                sort: false,
+            },
+        },
+        {
+            name: 'quota',
+            label: intl.formatMessage({
+                id: 'Admin.Throttling.Subscription.Throttling.policy.table.header.quota',
+                defaultMessage: 'Quota',
+            }),
+            options: {
+                filter: true,
+                sort: false,
+            },
+        },
+        {
+            name: 'unitTime',
+            label: intl.formatMessage({
+                id: 'Admin.Throttling.Subscription.Throttling.policy.table.header.unit.time',
+                defaultMessage: 'Unit Time',
+            }),
+            options: {
+                filter: true,
+                sort: false,
+            },
+        },
+        {
+            name: 'rateLimit',
+            label: intl.formatMessage({
+                id: 'Admin.Throttling.Subscription.Throttling.policy.table.header.rate.limit',
+                defaultMessage: 'Rate Limit',
+            }),
+            options: {
+                filter: true,
+                sort: false,
+            },
+        },
+        {
+            name: 'timeUnit',
+            label: intl.formatMessage({
+                id: 'Admin.Throttling.Subscription.Throttling.policy.table.header.time.unit',
+                defaultMessage: 'Time Unit',
+            }),
+            options: {
+                filter: true,
+                sort: false,
+            },
+        },
+    ];
+
+    const emptyBoxProps = {
+        content: (
+            <Typography variant='body2' color='textSecondary' component='p'>
+                <FormattedMessage
+                    id='AdminPages.Microgateways.List.empty.content.microgateways'
+                    defaultMessage={'It is possible to create a Microgateway distribution '
+                    + 'for a group of APIs. In order to group APIs, a label needs to be created'
+                    + ' and attached to the APIs that need to be in a single group.'}
+                />
+            </Typography>),
+        title: (
+            <Typography gutterBottom variant='h5' component='h2'>
+                <FormattedMessage
+                    id='AdminPages.Microgateways.List.empty.title.microgateways'
+                    defaultMessage='Microgateways'
+                />
+
+            </Typography>),
+    };
+
+    const addButtonOverride = (
+        <RouterLink to='/throttling/subscription/add'>
+            <Button variant='contained' color='primary'>
+                <FormattedMessage
+                    id='Throttling.Subscription.Policy.List.addButtonProps.title'
+                    defaultMessage='Add Policy'
+                />
+            </Button>
+        </RouterLink>
+    );
+
+    /**
+ * Mock API call
+ * @returns {Promise}.
+ */
+    function apiCall() {
+        let subscriptionThrottlingvalues;
+        return new Promise(((resolve) => {
```

Comment:
@kavishkafernando , Did we refactor the `ListBase` component to work with bare promise objects? rather than wrapping the API call promise with a new Promise?

Reply:
yes we can just return a promise rather than creating a new one.

Reply:
This is fixed